README.md

```diff
@@ -19,60 +19,93 @@ This is likely the quickest way to get the pipeline up and running and requires
-        ```
-        If a window appears, your computer is configured correctly. You only have to do this once.
-        
-        If stage 4 results in an error similar to `Error: Can't open display:`, then rerun step 4 with this modified command instead.
-        ```sh
-        docker run --rm -it -e DISPLAY=${DISPLAY} -v /tmp/.X11-unix:/tmp/.X11-unix centos:7 /bin/bash
-        yum install -y xorg-x11-apps
-        xeyes
-        ```
-        Now an window should appear.
-        
-   7. Clone this repository: 
-       ```
-       git clone https://github.com/LCOGT/lcogtsnpipe
-       ```
-   8. Build the Docker image:
-          ```
-          docker build -t lcogtsnpipe lcogtsnpipe
-          ```
-        
-   10. Set your environment variables to point to where you want to store data and catalogs.
+- Install [docker-compose](https://docs.docker.com/compose/install/)
+
+## System-specific configuration
+#### MacOS
+- Install [XQuartz](https://www.xquartz.org).
+- Install [socat](http://www.dest-unreach.org/socat/). If you have [Homebrew](https://brew.sh) installed, you can just run `brew install socat`.
+
+#### Linux Wayland
+- Install [socat](http://www.dest-unreach.org/socat/) with your package manager e.g. `sudo apt install socat`
+
+#### Linux X11
+- Modify your X11 config files to allow TCP connections: 
+     1. If you are running gdm or gdm3 for your display manager add the following to `/etc/gdm<3>/custom.conf`
+     ```
+     [security]
+     DisallowTCP=false
+     ```
+     2. If you are running lightdm, add the following to `/usr/share/lightdm/lightdm.conf.d/50-xserver-command.conf`
+     ```
+     [Seat:*]
+     xserver-command=X -core -listen tcp
+     ```
+     and the following to `/etc/lightdm/lightdm.conf`.
+     This file probably won't exist, you may create it if it is missing.
+     ```
+     [Seat:*]
+     xserver-allow-tcp=true
+     xserver-command=X -listen tcp
+     ```
+     3. Reboot your machine after updating the necessary config file.
+- Allow X11 connections: `xhost +local:docker`
+
+## Installation
+These instructions only need to be run once, when you set up the pipeline.  
+
+   1. Clone this repository: 
+      ```
+      git clone https://github.com/LCOGT/lcogtsnpipe
+      ```
+   2. Build the Docker image:
+      ```
+      docker build -t lcogtsnpipe lcogtsnpipe
+      ```
+   3. (MacOS only) Run this hack in the background to get the X11 forwarding to work:
+      ```
+      socat TCP-LISTEN:6000,reuseaddr,fork UNIX-CLIENT:\"$DISPLAY\"
+      ```
+      This will block the current terminal. Open a new terminal/tab and run
+      ```
+      docker run --rm -it -e DISPLAY=${LCOSNDISPLAY:-host.docker.internal:0} lcogtsnpipe /bin/bash
+      xeyes
+      ```
+      If a window appears, your computer is configured correctly. `exit` the container.
+
+   4. (Linux Wayland only) Run the following command for X11 forwarding:
+      ```
+      socat TCP-LISTEN:6000,reuseaddr,fork UNIX-CONNECT:/tmp/.X11-unix/X0
```

Comment:
Isn't this identical for Mac? We can say mac or wayland here if so (not only linux wayland).

Reply:
Hi curtis,
For MacOS it's `UNIX-CLIENT:\"$DISPLAY\"` while for wayland it's `UNIX-CONNECT:/tmp/.X11-unix/X0`

Reply:
Btw have you tried this? 
Ideally we wouldn't need `socat` for wayland, but after scouring countless forums, it was the only solution that worked.

Reply:
Oh ok. Could you make a note there for mac as welI basically saying just what you did here. I don't have a system with wayland to test. I feel like every year x11 gets a little harder to support.

Reply:
Okay. Just to confirm, it would mean combining point 3 and 4, right?

Also, yes. Ubuntu and fedora both now has wayland as the default.